api/include/opentelemetry/logs/event_id.h

```diff
@@ -0,0 +1,35 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#ifdef ENABLE_LOGS_PREVIEW
+
+#  include "opentelemetry/nostd/unique_ptr.h"
+#  include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace logs
+{
+
+/**
+ * EventId class which acts the Id of the event with an optional name.
+ */
+class EventId
+{
+public:
+  EventId(int64_t id, nostd::string_view name) noexcept
+  {
+    id_   = id;
+    name_ = nostd::unique_ptr<char[]>{new char[name.length() + 1]};
+    std::copy(name.begin(), name.end(), name_.get());
+  }
+
+public:
+  int64_t id_;
+  nostd::unique_ptr<char[]> name_;
```

Comment:
Why not use `std::string name_;` here? The usage of this variable above seems just like std::string, and do not add a tail `\0` ?

Reply:
`EventId` can be read from both user and SDK code, which requires it to be ABI compatible. `std::string` cannot guarantee such ABI compatibility requirement across the boundary between user code/SDK so cannot be used. I'd expect there is `nostd::string` but we don't have it, then switch to primitive `char` pointer wrapped in `nostd::unique_ptr`. Let me know if this addresses the concern.

Reply:
Yes, that makes sense. Set the last element of `name_` as 0.